packages/tupaia-web/src/features/Map/MapOverlays/DataVisualsLayer.tsx

```diff
@@ -49,8 +50,11 @@ export const DataVisualsLayer = ({
   return (
     <LayerGroup>
       {measureData.map((measure: MeasureData) => {
-        const { region, organisationUnitCode: entity, color, name } = measure;
+        const { region, organisationUnitCode: entity, color, name, code } = measure;
         if (region) {
+          if (code === entityCode) {
+            return <ActiveEntityPolygon key={entity} entity={measure} />; // this is so that the polygon is displayed as the active entity, i.e correctlyt shaded etc.
+          }
```

Comment:
I was conflicted whether to display this in here or just make it return null so that the `PolygonNavigationLayer` handled it. But if we did that then would have to fetch the measure item to get the color anyway

Reply:
Yes I know what you mean about not knowing whether to put things in here or PolygonNavigationLayer. I am open to suggestions about organising that better if possible but otherwise this seems good.